### PR TITLE
Add Switch component as a variant to Checkbox

### LINKED ISSUE #745: Support Switch component
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Summary 💡

https://mui.com/material-ui/react-switch/

One of: https://github.com/mui/mui-toolpad/issues/78

### PR DESCRIPTION
**Pull request by JerryWu1234**
Closes #745

- [x] I've read and followed the [contributing guide](https://github.com/mui/mui-toolpad/blob/master/CONTRIBUTING.md#sending-a-pull-request) on how to create great pull requests.
- [x] I've updated the relevant documentation for any new or updated feature.
- [x] I've linked relevant GitHub issue with "Closes #<issue id>".
- [x] I've added a visual demonstration in the form of a screenshot or video.

### CODE REVIEW

**Comment by JerryWu1234** [comment]
image
image
image

**Comment by JerryWu1234** [comment]
@Janpot i done this

File: packages/toolpad-components/src/Checkbox.tsx [hunk 1]
```diff
@@ -52,42 +73,73 @@ function Checkbox({ ...rest }: CheckboxProps) {
...
+      },
+      default: 'checkBox',
+      label: 'Mode',
+      control: { type: 'ToggleButtons' },
+    },
     label: {
       helperText: 'A text or an element to be used in an enclosing label element.',
       type: 'string',
-      default: 'Checkbox',
+      default: 'test',
```
Comment (Janpot) [inline packages/toolpad-components/src/Checkbox.tsx:137]:
```suggestion
      default: 'Checkbox',
```

File: packages/toolpad-components/src/Checkbox.tsx [hunk 2]
```diff
@@ -118,10 +168,12 @@ export default createComponent(FormWrappedCheckbox, {
     fullWidth: {
       helperText: 'Whether the select should occupy all available horizontal space.',
       type: 'boolean',
+      visible: ({ mode }: Props) => mode === 'checkBox',
```
Comment (Janpot) [inline packages/toolpad-components/src/Checkbox.tsx:171]: Why doesn't this apply to the switch?

File: test/integration/backend-basic/fixture/toolpad/pages/basic/page.yml [hunk 3]
```diff
@@ -151,6 +151,10 @@ spec:
           $$jsExpressionAction: |-
             await setCustomCookie.call()
             context.refetch()
+    - component: Checkbox
```
Comment (Janpot) [inline test/integration/backend-basic/fixture/toolpad/pages/basic/page.yml:154]: This is not showing up in the screenshot. I don't believe the visual regression tests are using this fixture

Comment (JerryWu1234) [reply test/integration/backend-basic/fixture/toolpad/pages/basic/page.yml:154]: I follow this [link](https://github.com/mui/mui-toolpad/pull/2494#issuecomment-1698071572) 
and I drag a componen into the editor

Comment (Janpot) [reply test/integration/backend-basic/fixture/toolpad/pages/basic/page.yml:154]: Are you sure you used that exact command? The change in this PR is under test/integration/backend-basic, I'd expect it to be under test/visual/components.

File: packages/toolpad-components/src/Checkbox.tsx [hunk 4]
```diff
@@ -63,6 +63,19 @@ function Checkbox({ ...rest }: Props) {
     [onFormInputChange],
   );
 
+  const props = React.useMemo(
```
Comment (Janpot) [inline packages/toolpad-components/src/Checkbox.tsx:66]: The `useMemo` is not necessary here, we're spreading these props to a component anyway.

**Review by Janpot — APPROVED** [review]
Thank you, I made a few tweaks before merging:

* Use more neutral label "Label" instead of "Checkbox"
* Pass `checked` property to the MUI component
* Run the visual regression test app and use the UI to update the Switch and Checkbox labels

**Comment by JerryWu1234** [comment]
@Janpot hi Janpot

How can I join your team? because I want to be exposed to an English-speaking environment more and also discuss some of my preliminary ideas. On the other hand, I hope to use GitHub Copilot. Thank you so much in advance

**Comment by Janpot** [comment]
@JerryWu1234 Thank you for your interest. We currently don't have any open positions for the toolpad team. When a job opening becomes available it will be announced on our [careers page](https://mui.com/careers/#open-roles).

**Comment by JerryWu1234** [comment]
> @JerryWu1234 Thank you for your interest. We currently don't have any open positions for the toolpad team. When a job opening becomes available it will be announced on our [careers page](https://mui.com/careers/#open-roles).

image

Is it only regular staff who can join this group as maintainer ? 
is there any opportunity for me to join the group without being a regular staff member and without having to pay.
but I can participate your meet.

**Comment by Janpot** [comment]
As far as I can see these are either employees, [ex-employees](https://github.com/orgs/mui/teams/emeriti) or members of the project from before the incorporation. The latter don't have write access to repositories, no external contributors attend meetings.

**Pull request merged**